File: tests/create_test_image.cpp
/**
 * Create a simple test image for the image processing server tests
 * Uses OpenCV to generate a test image
 */

#include <opencv2/opencv.hpp>
#include <iostream>

int main() {
    // Create a 256x256 RGB image with a blue background
    cv::Mat img(256, 256, CV_8UC3, cv::Scalar(73, 109, 137));
    
    // Draw a yellow rectangle
    cv::rectangle(img, cv::Point(50, 50), cv::Point(200, 150), cv::Scalar(0, 255, 255), -1);
    
    // Draw a red ellipse
    cv::ellipse(img, cv::Point(150, 150), cv::Size(50, 50), 0, 0, 360, cv::Scalar(0, 0, 255), -1);
    
    // Draw a green line
    cv::line(img, cv::Point(0, 0), cv::Point(256, 256), cv::Scalar(0, 255, 0), 5);
    
    // Save the image
    bool success = cv::imwrite("test_image.jpg", img);
    
    if (success) {
        std::cout << "Created test_image.jpg" << std::endl;
        return 0;
    } else {
        std::cerr << "Failed to create test image" << std::endl;
        return 1;
    }
}

File: tests/create_test_image.py
#!/usr/bin/env python3
"""
Create a simple test image for the image processing server tests
"""

from PIL import Image, ImageDraw
import numpy as np

def create_test_image():
    # Create a simple test image (256x256 RGB image with shapes)
    img = Image.new('RGB', (256, 256), color=(73, 109, 137))
    d = ImageDraw.Draw(img)
    
    # Draw a yellow rectangle
    d.rectangle([50, 50, 200, 150], fill=(255, 255, 0))
    
    # Draw a red ellipse
    d.ellipse([100, 100, 200, 200], fill=(255, 0, 0))
    
    # Draw a blue line
    d.line([0, 0, 256, 256], fill=(0, 0, 255), width=5)
    
    # Save the image
    img.save('test_image.jpg', 'JPEG')
    print("Created test_image.jpg")

if __name__ == "__main__":
    create_test_image()